fe/fe-core/src/main/java/com/starrocks/http/rest/v2/TablePartitionAction.java

```diff
@@ -0,0 +1,93 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.http.rest.v2;
+
+
+import com.starrocks.catalog.OlapTable;
+import com.starrocks.catalog.Partition;
+import com.starrocks.http.ActionController;
+import com.starrocks.http.BaseRequest;
+import com.starrocks.http.BaseResponse;
+import com.starrocks.http.IllegalArgException;
+import com.starrocks.http.rest.v2.RestBaseResultV2.PagedResult;
+import com.starrocks.http.rest.v2.vo.TablePartitionView;
+import com.starrocks.privilege.AccessDeniedException;
+import io.netty.handler.codec.http.HttpMethod;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.BooleanUtils;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static com.starrocks.catalog.InternalCatalog.DEFAULT_INTERNAL_CATALOG_NAME;
+
+public class TablePartitionAction extends TableBaseAction {
+
+    private static final String TEMPORARY_KEY = "temporary";
+
+    public TablePartitionAction(ActionController controller) {
+        super(controller);
+    }
+
+    public static void registerAction(ActionController controller) throws IllegalArgException {
+        controller.registerHandler(
+                HttpMethod.GET,
+                String.format(
+                        "/api/v2/catalogs/{%s}/databases/{%s}/tables/{%s}/partition", CATALOG_KEY, DB_KEY, TABLE_KEY),
+                new TablePartitionAction(controller));
+    }
+
+    @Override
+    protected void doExecuteWithoutPassword(
+            BaseRequest request, BaseResponse response) throws AccessDeniedException {
+        String catalogName = getSingleParameterOrDefault(
+                request, CATALOG_KEY, DEFAULT_INTERNAL_CATALOG_NAME, Function.identity());
+        String dbName = getSingleParameterRequired(request, DB_KEY, Function.identity());
+        String tableName = getSingleParameterRequired(request, TABLE_KEY, Function.identity());
+        boolean temporary = getSingleParameterOrDefault(request, TEMPORARY_KEY, false, BooleanUtils::toBoolean);
+        int pageNum = getPageNum(request);
+        int pageSize = getPageSize(request);
+
+        PagedResult<TablePartitionView> pagedResult = new PagedResult<>();
+        pagedResult.setPageNum(pageNum);
+        pagedResult.setPageSize(pageSize);
+        pagedResult.setPages(0);
+        pagedResult.setTotal(0);
+        pagedResult.setItems(new ArrayList<>(0));
+
+        OlapTable olapTable = getOlapTable(catalogName, dbName, tableName);
+        Collection<Partition> partitions =
+                temporary ? olapTable.getTempPartitions() : olapTable.getPartitions();
```

Comment:
Do we need to lock the table to get a partition snapshot?
Or will the olapTable be modified when geting partitions.

Reply:
Yeah, It is a problem, and i encapsulated the processing logic of OlapTable into the db read lock.